### ISSUE TITLE
DL4J: Add CuDNN BatchNorm rank 2 support

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#6015]
CuDNN doesn't support rank 2 for batch norm. But, we can simply reshape from [mb,d] to [mb,d,1,1] and use the CuDNN implementation. This should be faster than our implementation.
